samza-core/src/main/java/org/apache/samza/operators/impl/SendUpdateToTableOperatorImpl.java

```
@@ -0,0 +1,93 @@
+ * @param <U> the type of the update applied to this table
+ */
+public class SendUpdateToTableOperatorImpl<K, V, U>
+    extends OperatorImpl<KV<K, UpdateMessage<U, V>>, KV<K, UpdateMessage<U, V>>> {
+  private final SendUpdateToTableOperatorSpec<K, V, U> sendUpdateToTableOpSpec;
+  private final ReadWriteTable<K, V, U> table;
+
+  public SendUpdateToTableOperatorImpl(SendUpdateToTableOperatorSpec<K, V, U>  sendUpdateToTableOpSpec, Context context) {
+    this.sendUpdateToTableOpSpec = sendUpdateToTableOpSpec;
+    this.table = context.getTaskContext().getTable(sendUpdateToTableOpSpec.getTableId());
+  }
+
+  @Override
+  protected void handleInit(Context context) {
+  }
+
+  @Override
+  protected CompletionStage<Collection<KV<K, UpdateMessage<U, V>>>> handleMessageAsync(KV<K, UpdateMessage<U, V>> message,
+      MessageCollector collector, TaskCoordinator coordinator) {
+    final CompletableFuture<Void> updateFuture = table.updateAsync(message.getKey(), message.getValue().getUpdate(),
+        sendUpdateToTableOpSpec.getArgs());
+    return updateFuture
+        .handle((result, ex) -> {
+          // RecordDoesNotExistException is wrapped in java.util.concurrent.CompletionException
+          return ex != null
+              && ex.getCause() instanceof RecordDoesNotExistException
+              && message.getValue().getDefault() != null;
+        })
+        .thenCompose(shouldPutDefault -> {
+          // If update fails for a given key due to a RecordDoesNotExistException exception thrown and a default is
+          // provided, then attempt to PUT a default record for the key and then apply the update
+          if (shouldPutDefault) {
+            final CompletableFuture<Void> putFuture = table.putAsync(message.getKey(), message.getValue().getDefault(),
+                sendUpdateToTableOpSpec.getArgs());
+            return putFuture.thenCompose(res -> table.updateAsync(message.getKey(), message.getValue().getUpdate(),
```

we might want to also ignore the exception during putFuture. You can add something like:
```
putFuture.exceptionally(e -> {
    log exception;
    return null;
}).thenCompose(....)
```

---

Done.